### PR TITLE
Add pagination and server-side search

### PR DESCRIPTION
**Pull request by teonaandreea**
- Adds pagination & server-side search
- Fix bug that occurred when dragging the chart annotation, caused by missing unique labels for each of the lines

### CODE REVIEW

File: frontend/src/pages/ModelsList/ModelsList.js [hunk 1]
```diff
@@ -10,18 +10,67 @@ class ModelsList extends Component {
...
+            // fetch data for that page
+            fetch(baseUrl, {
+                accept: 'application/json',
+            })
+                .then(results => results.json())
+                .then(data => {
+                    this.setState({ models: data.models, loading: false })
+                })
+                .catch(err => console.log(err));
+        } else if (state.filtered.length > 0) {
```
Comment (rasmusbergpalm) [inline frontend/src/pages/ModelsList/ModelsList.js]: Does this mean I can't paginate a search? e.g. I search for a user, then paginate through his models?

File: frontend/src/pages/ModelsList/ModelsList.js [hunk 2]
```diff
@@ -10,18 +10,67 @@ class ModelsList extends Component {
...
+                searchUrl += `&${filter.id}=${filter.value}`;
+            });
+
+            if(this.timeout) {clearTimeout(this.timeout)}
+            this.timeout = setTimeout(() => {
+                // perform search
+                fetch(searchUrl, {
+                    accept: 'application/json',
+                })
+                    .then(results => results.json())
```
Comment (rasmusbergpalm) [inline frontend/src/pages/ModelsList/ModelsList.js]: There's a possible edge case due to the async nature of JS where 

client searches for a
client searches for ab
results for ab returns
results for a returns

so the result is the filter says 'ab' but the results are from 'a'.

I'm not sure it's worth protecting against though

File: frontend/src/utils/chartConfigData.js [hunk 3]
```diff
@@ -162,14 +168,6 @@ export const getChartOptions = (threshold, savedThreshold, handleOnDrag) => ({
     legend: {
         display: false,
     },
-    // layout: {
```
Comment (rasmusbergpalm) [inline frontend/src/utils/chartConfigData.js]: kill zombie code

Comment (teonaandreea) [reply frontend/src/utils/chartConfigData.js]: that was already removed 👍

File: frontend/src/pages/ModelsList/ModelsList.js (hunk 1 above)
Comment (teonaandreea) [reply frontend/src/pages/ModelsList/ModelsList.js]: the searches are paginated too. 

when you start typing into the search field, the value is saved in this `state.filtered` array. and the table library also sets the page to 0 by itself whenever you start typing in one of the searches. the `baseUrl` that contains the page number is set above the if statements, so the page is always the correct one.

eg. user types into search field for the first time > `fetchData` is called > state.page is being set to 0 by the library > baseUrl has 0 as a page > state.filtered contains the value that the user wrote > calling the API with search terms > setting `totalPages` to the new value, that represents how many pages we have for that search

Once we have a search term + user clicks "next page" > `fetchData` is called > state.page increases with 1 > baseUrl has 1 as a page > state.filtered still contains the value that the user wrote > calling the API with search terms

Comment (teonaandreea) [reply frontend/src/pages/ModelsList/ModelsList.js]: I think me setting values for the page in the react component state is a bit confusing, and it's actually not really needed, so I'll make a change to remove those 👌

File: frontend/src/pages/ModelsList/ModelsList.js (hunk 2 above)
Comment (teonaandreea) [reply frontend/src/pages/ModelsList/ModelsList.js]: yeah, I know 🤔 I'll have a think and see if there's anything I can do about it..

File: frontend/src/pages/ModelsList/ModelsList.js (hunk 1 above)
Comment (rasmusbergpalm) [reply frontend/src/pages/ModelsList/ModelsList.js]: Ok. Thanks!

File: frontend/src/pages/ModelsList/ModelsList.js (hunk 2 above)
Comment (rasmusbergpalm) [reply frontend/src/pages/ModelsList/ModelsList.js]: Maybe you can store the latest XHR request object when doing the request and then check if the response belongs to it when it comes back. Anyways, it's probably not super important. Also, someone must have solved this problem for react a hundred times over :)

File: frontend/src/utils/chartConfigData.js (hunk 3 above)
Comment (rasmusbergpalm) [reply frontend/src/utils/chartConfigData.js]: 👍

File: frontend/src/pages/ModelsList/ModelsList.js (hunk 2 above)
Comment (rasmusbergpalm) [reply frontend/src/pages/ModelsList/ModelsList.js]: Something like this https://stackoverflow.com/a/42151219/9862604

**Pull request merged**